[2025-04-22 16:17:27,881][flwr][WARNING] - DEPRECATED FEATURE: flwr.simulation.start_simulation() is deprecated.
	Instead, use the `flwr run` CLI command to start a local simulation in your Flower app, as shown for example below:

		$ flwr new  # Create a new Flower app from a template

		$ flwr run  # Run the Flower app in Simulation Mode

	Using `start_simulation()` is deprecated.

            This is a deprecated feature. It will be removed
            entirely in future versions of Flower.
        
[2025-04-22 16:17:27,881][flwr][INFO] - Starting Flower simulation, config: num_rounds=200, no round_timeout
[2025-04-22 16:17:32,243][flwr][INFO] - Flower VCE: Ray initialized with resources: {'accelerator_type:G': 1.0, 'node:__internal_head__': 1.0, 'CPU': 16.0, 'memory': 13248984269.0, 'node:127.0.0.1': 1.0, 'object_store_memory': 5678136115.0, 'GPU': 1.0}
[2025-04-22 16:17:32,243][flwr][INFO] - Optimize your simulation with Flower VCE: https://flower.ai/docs/framework/how-to-run-simulations.html
[2025-04-22 16:17:32,244][flwr][INFO] - Flower VCE: Resources for each Virtual Client: {'num_cpus': 2, 'num_gpus': 0.125}
[2025-04-22 16:17:32,251][flwr][INFO] - Flower VCE: Creating VirtualClientEngineActorPool with 8 actors
[2025-04-22 16:17:32,252][flwr][INFO] - [INIT]
[2025-04-22 16:17:32,252][flwr][INFO] - Using initial global parameters provided by strategy
[2025-04-22 16:17:32,252][flwr][INFO] - Starting evaluation of initial global parameters
[2025-04-22 16:17:40,774][flwr][INFO] - initial parameters (loss, other metrics): 0.03620454158782959, {'accuracy': 0.1}
[2025-04-22 16:17:40,774][flwr][INFO] - 
[2025-04-22 16:17:40,774][flwr][INFO] - [ROUND 1]
[2025-04-22 16:17:40,774][flwr][INFO] - configure_fit: strategy sampled 16 clients (out of 100)
[2025-04-22 16:17:47,999][flwr][ERROR] - Traceback (most recent call last):
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 89, in _submit_job
    self.actor_pool.submit_client_job(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 247, in submit_client_job
    self.submit(actor_fn, job)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 225, in submit
    future = fn(actor, app_fn, mssg, cid, context)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 90, in <lambda>
    lambda a, a_fn, mssg, partition_id, context: a.run.remote(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 216, in remote
    return self._remote(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\auto_init_hook.py", line 21, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\tracing\tracing_helper.py", line 422, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 376, in _remote
    return invocation(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 357, in invocation
    return actor._actor_method_call(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 1518, in _actor_method_call
    object_refs = worker.core_worker.submit_actor_task(
  File "python\\ray\\_raylet.pyx", line 3993, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 3998, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 870, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 861, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 911, in ray._raylet.prepare_args_internal
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 556, in serialize
    return self._serialize_to_msgpack(value)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 534, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 481, in _serialize_to_pickle5
    raise e
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 476, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1479, in dumps
    cp.dump(obj)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1245, in dump
    return super().dump(obj)
MemoryError

[2025-04-22 16:17:47,999][flwr][ERROR] - Traceback (most recent call last):
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 95, in _submit_job
    out_mssg, updated_context = self.actor_pool.get_client_result(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 394, in get_client_result
    self.process_unordered_future(timeout=timeout)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 376, in process_unordered_future
    self._return_actor(actor)  # type: ignore
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\actor_pool.py", line 375, in _return_actor
    self.submit(*self._pending_submits.pop(0))
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 225, in submit
    future = fn(actor, app_fn, mssg, cid, context)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 90, in <lambda>
    lambda a, a_fn, mssg, partition_id, context: a.run.remote(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 216, in remote
    return self._remote(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\auto_init_hook.py", line 21, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\tracing\tracing_helper.py", line 422, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 376, in _remote
    return invocation(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 357, in invocation
    return actor._actor_method_call(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 1518, in _actor_method_call
    object_refs = worker.core_worker.submit_actor_task(
  File "python\\ray\\_raylet.pyx", line 3993, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 3998, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 870, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 861, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 911, in ray._raylet.prepare_args_internal
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 556, in serialize
    return self._serialize_to_msgpack(value)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 534, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 481, in _serialize_to_pickle5
    raise e
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 476, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1479, in dumps
    cp.dump(obj)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1245, in dump
    return super().dump(obj)
MemoryError

[2025-04-22 16:17:47,999][flwr][ERROR] - Traceback (most recent call last):
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 89, in _submit_job
    self.actor_pool.submit_client_job(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 247, in submit_client_job
    self.submit(actor_fn, job)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 225, in submit
    future = fn(actor, app_fn, mssg, cid, context)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 90, in <lambda>
    lambda a, a_fn, mssg, partition_id, context: a.run.remote(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 216, in remote
    return self._remote(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\auto_init_hook.py", line 21, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\tracing\tracing_helper.py", line 422, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 376, in _remote
    return invocation(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 357, in invocation
    return actor._actor_method_call(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 1518, in _actor_method_call
    object_refs = worker.core_worker.submit_actor_task(
  File "python\\ray\\_raylet.pyx", line 3993, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 3998, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 870, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 861, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 911, in ray._raylet.prepare_args_internal
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 556, in serialize
    return self._serialize_to_msgpack(value)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 534, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 481, in _serialize_to_pickle5
    raise e
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 476, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1479, in dumps
    cp.dump(obj)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1245, in dump
    return super().dump(obj)
MemoryError

[2025-04-22 16:17:47,999][flwr][ERROR] - 
[2025-04-22 16:17:48,000][flwr][ERROR] - Traceback (most recent call last):
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 95, in _submit_job
    out_mssg, updated_context = self.actor_pool.get_client_result(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 394, in get_client_result
    self.process_unordered_future(timeout=timeout)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 376, in process_unordered_future
    self._return_actor(actor)  # type: ignore
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\actor_pool.py", line 375, in _return_actor
    self.submit(*self._pending_submits.pop(0))
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 225, in submit
    future = fn(actor, app_fn, mssg, cid, context)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 90, in <lambda>
    lambda a, a_fn, mssg, partition_id, context: a.run.remote(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 216, in remote
    return self._remote(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\auto_init_hook.py", line 21, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\tracing\tracing_helper.py", line 422, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 376, in _remote
    return invocation(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 357, in invocation
    return actor._actor_method_call(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 1518, in _actor_method_call
    object_refs = worker.core_worker.submit_actor_task(
  File "python\\ray\\_raylet.pyx", line 3993, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 3998, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 870, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 861, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 911, in ray._raylet.prepare_args_internal
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 556, in serialize
    return self._serialize_to_msgpack(value)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 534, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 481, in _serialize_to_pickle5
    raise e
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 476, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1479, in dumps
    cp.dump(obj)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1245, in dump
    return super().dump(obj)
MemoryError

[2025-04-22 16:17:48,005][flwr][ERROR] - 
[2025-04-22 16:17:48,005][flwr][ERROR] - 
[2025-04-22 16:17:48,006][flwr][ERROR] - 
[2025-04-22 16:17:50,819][flwr][ERROR] - Traceback (most recent call last):
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 95, in _submit_job
    out_mssg, updated_context = self.actor_pool.get_client_result(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 394, in get_client_result
    self.process_unordered_future(timeout=timeout)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 376, in process_unordered_future
    self._return_actor(actor)  # type: ignore
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\actor_pool.py", line 375, in _return_actor
    self.submit(*self._pending_submits.pop(0))
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_actor.py", line 225, in submit
    future = fn(actor, app_fn, mssg, cid, context)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\flwr\simulation\ray_transport\ray_client_proxy.py", line 90, in <lambda>
    lambda a, a_fn, mssg, partition_id, context: a.run.remote(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 216, in remote
    return self._remote(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\auto_init_hook.py", line 21, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\util\tracing\tracing_helper.py", line 422, in _start_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 376, in _remote
    return invocation(args, kwargs)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 357, in invocation
    return actor._actor_method_call(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\actor.py", line 1518, in _actor_method_call
    object_refs = worker.core_worker.submit_actor_task(
  File "python\\ray\\_raylet.pyx", line 3993, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 3998, in ray._raylet.CoreWorker.submit_actor_task
  File "python\\ray\\_raylet.pyx", line 870, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 861, in ray._raylet.prepare_args_and_increment_put_refs
  File "python\\ray\\_raylet.pyx", line 911, in ray._raylet.prepare_args_internal
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 556, in serialize
    return self._serialize_to_msgpack(value)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 534, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 481, in _serialize_to_pickle5
    raise e
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\_private\serialization.py", line 476, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1479, in dumps
    cp.dump(obj)
  File "D:\flower\baselines\fedpara\venv\lib\site-packages\ray\cloudpickle\cloudpickle.py", line 1245, in dump
    return super().dump(obj)
MemoryError

[2025-04-22 16:17:50,819][flwr][ERROR] - 
